### PR TITLE
Add audit trail support

### PR DESCRIPTION
**Pull request by gargravarr**
This update adds support for audit event to be sent to Zalando's AuditTrail. What's left to implement is support for writing events to an s3 Bucket when it is not possible to communicate with the audit system. It will be addressed in other Pull Request.

### CODE REVIEW

File: audittrail.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        self.__send_to_audit_trail(event_id, event["event_type"]["name"], payload)
+
+    def __send_to_audit_trail(self, event_id, event_name, payload):
+        headers = {
+            "Content-type": "application/json",
+            "Authorization": "Bearer {}".format(self.token_source.get(self.token_name)),
+        }
+
+        try:
+            r = self.session.put(self.url.format(event_id), data=payload, headers=headers)
```
Comment (aermakov-zalando) [inline audittrail.py]: You can do `json=payload` and then you don't need to set the content-type manually.

**Comment by aermakov-zalando** [comment]
👍

File: audittrail.py (hunk 1 above)
Comment (hjacobs) [inline audittrail.py]: HTTP calls should have a timeout :smile: I see that I also forgot this in other calls of this project :man_facepalming:

**Comment by aermakov-zalando** [comment]
👍

**Comment by gargravarr** [comment]
:+1:

**Pull request merged**